devicetypes/smartthings/eaton-accesory-dimmer.src/eaton-accessory-dimmer.groovy

```diff
@@ -0,0 +1,286 @@
+/**
+ *  Eaton Accessory Dimmer
+ *
+ *  Copyright 2017 Smartthings
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *  	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+metadata {
+	definition (name: "Eaton Accessory Dimmer", namespace: "smartthings", author: "Smartthings") {
+		capability "Actuator"
+		capability "Sensor"
+		capability "Switch"
+		capability "Switch Level"
+		capability "Refresh"
+		fingerprint mfr:"001A",prod:"4441", model:"0000", deviceJoinName: "RF9542-Z - RF Accessory Dimmer"
+		// zw:L type:1202 mfr:001A prod:4441 model:0000 ver:3.17 zwv:3.67 lib:03 cc:26,27,75,86,70,71,85,77,2B,2C,72,73,87
+	}
+	
+	tiles(scale: 2) {
+		multiAttributeTile(name:"switch", type: "lighting", width: 6, height: 4, canChangeIcon: true) {
+			tileAttribute ("device.switch", key: "PRIMARY_CONTROL") {
+				attributeState "on",  label:'${name}', action:"switch.off", icon:"st.switches.switch.on",  backgroundColor: "#00a0dc"
+				attributeState "off", label:'${name}', action:"switch.on",  icon:"st.switches.switch.off", backgroundColor:"#ffffff"
+			}
+			tileAttribute ("device.level", key: "SLIDER_CONTROL") {
+				attributeState "level", action:"switch level.setLevel", range: "1..100",  unit:"%"
+			}
+		}
+		standardTile("refresh", "device.switch", inactiveLabel: false, decoration: "flat", width: 6, height: 2, backgroundColor: "#00a0dc") {
+			state "default", label: '', action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+
+		main("switch")
+		details(["switch","refresh"])
+	}
+
+	preferences {
+		input( "powerUpState", "enum", title: "Power Up State", description: "Power Up State",
+				required: false, options:["on": "On", "off": "Off", "lastState": "Last State"]
+		)
+		input( "delayedOffTime", "number",
+				title: "Time after which switch will turn off when using delayed off feature [seconds]",
+				description: "[1..255]", range: "1..255", required: false, displayDuringSetup: true
+		)
+		input( "dimmerRampTime", "number",
+				title: "Time of dimmed switching off [seconds]",
```

Comment:
I think this needs to be addressed because of how the dimmer value is treated.

If I read the comments right, values between 0-127 are in seconds, while values between 128-255 are minutes?

I propose that we just limit the user to seconds (1-127) to avoid confusion.

Reply:
This configutration value is designed by manufacturer, and it is 0-255 seconds.

Comments regarding minutes are part of setLevel(value, duration) method. Those are handling Z-Wave specification of zwave.switchMultilevelV2.switchMultilevelSet(value, dimmingDuration) command, which sets dimmer ramp time during request. Normal switchMultilevelSet(value), which is called in setLevel(value) command uses default dimmer ramp time taken from device's preferences.

Reply:
Okay. Got it. Thanks for the clarification.